chia/plotters/madmax.py

```diff
@@ -0,0 +1,246 @@
+import asyncio
+import traceback
+import subprocess
+import os
+import logging
+import sys
+
+from pathlib import Path
+from typing import Any, Dict, Optional
+from chia.plotting.create_plots import resolve_plot_keys
+
+log = logging.getLogger(__name__)
+
+
+MADMAX_PLOTTER_DIR = "madmax-plotter"
+
+
+def is_madmax_supported() -> bool:
+    return sys.platform.startswith("linux") or sys.platform.startswith("darwin")
+
+
+def get_madmax_install_path(plotters_root_path: Path) -> Path:
+    return plotters_root_path / MADMAX_PLOTTER_DIR
+
+
+def get_madmax_executable_path(plotters_root_path: Path) -> Path:
+    return get_madmax_install_path(plotters_root_path) / "build/chia_plot"
+
+
+def get_madmax_install_info(plotters_root_path: Path) -> Optional[Dict[str, Any]]:
+    info: Dict[str, Any] = {"display_name": "madMAx Plotter"}
+    installed: bool = False
+    supported: bool = is_madmax_supported()
+
+    if get_madmax_executable_path(plotters_root_path).exists():
+        installed = True
+        # TODO: Figure out how to get madmax version
+
+    info["installed"] = installed
+    if installed is False:
+        info["can_install"] = supported
+
+    return info
+
+
+def install_madmax(plotters_root_path: Path):
+    if is_madmax_supported():
+        print("Installing dependencies.")
+        if sys.platform.startswith("linux"):
+            try:
+                subprocess.run(
+                    [
+                        "sudo",
+                        "apt",
+                        "install",
+                        "-y",
+                        "libsodium-dev",
+                        "cmake",
+                        "g++",
+                        "git",
+                        "build-essential",
+                    ]
+                )
+            except Exception as e:
+                raise ValueError(f"Could not install dependencies. {e}")
+        if sys.platform.startswith("darwin"):
+            try:
+                subprocess.run(
+                    [
+                        "brew",
+                        "install",
+                        "libsodium",
+                        "cmake",
+                        "git",
+                        "autoconf",
+                        "automake",
+                        "libtool",
+                        "wget",
+                    ]
+                )
+                subprocess.run(["brew", "link", "cmake"])
+            except Exception as e:
+                raise ValueError(f"Could not install dependencies. {e}")
+
+        try:
+            subprocess.run(["git", "--version"])
+        except FileNotFoundError as e:
+            raise ValueError(f"Git not installed. Aborting madmax install. {e}")
```

Comment:
In general we seem to also catch 'all' errors.  For consistency...

```suggestion
        except FileNotFoundError as e:
            raise ValueError(f"Git not installed. Aborting madmax install. {e}")
        except Exception as e:
            raise ValueError(f"Error checking Git version. {e}")
```